lanarky/callbacks/llm.py

```diff
@@ -22,25 +23,57 @@ class AsyncLLMChainStreamingCallback(AsyncStreamingResponseCallback):
 
     async def on_llm_new_token(self, token: str, **kwargs: Any) -> None:
         """Run on new LLM token. Only available when streaming is enabled."""
+        self.llm_cache_used = (
+            False  # cache missed (or was never enabled) if we are here
+        )
```

Comment:
```suggestion
        if self.llm_cache_used: # cache missed (or was never enabled) if we are here
            self.llm_cache_used = False
```

minor optimisation. can be applied elsewhere too.